File: frontend/src/types/report.ts
export type ReportPeriod = "day" | "week" | "month" | "year";

export interface CategorySummary {
  category: string;
  count: number;
  total_amount: number;
}

export interface PeriodReport {
  period: ReportPeriod;
  period_label: string;
  start_date: string;
  end_date: string;
  total_invoices: number;
  total_amount: number;
  paid_invoices: number;
  unpaid_invoices: number;
  total_paid_amount: number;
  matched_invoices: number;
  unmatched_invoices: number;
  needs_review: number;
  bank_transactions: number;
  bank_matched: number;
  bank_needs_review: number;
  by_category: CategorySummary[];
}

export interface PeriodReportParams {
  period: ReportPeriod;
  anchor_date?: string;
}

export const REPORT_PERIOD_OPTIONS: { value: ReportPeriod; label: string }[] = [
  { value: "day", label: "Daily" },
  { value: "week", label: "Weekly" },
  { value: "month", label: "Monthly" },
  { value: "year", label: "Yearly" },
];

File: backend/schemas/report.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, Field


class CategorySummary(BaseModel):
    category: str
    count: int
    total_amount: Decimal = Field(default=Decimal("0"))


class PeriodReportResponse(BaseModel):
    period: str
    period_label: str
    start_date: date
    end_date: date
    total_invoices: int
    total_amount: Decimal
    paid_invoices: int
    unpaid_invoices: int
    total_paid_amount: Decimal
    matched_invoices: int
    unmatched_invoices: int
    needs_review: int
    bank_transactions: int
    bank_matched: int
    bank_needs_review: int
    by_category: list[CategorySummary]
